packages/api/tests/services/dbt.test.ts

```diff
@@ -0,0 +1,227 @@
+import { createDatabaseCon } from '../../src/clients/db/orm'
+import { DbtBlock } from '../../src/core/block/dbt'
+import { QuestionBlock } from '../../src/core/block/question'
+import { FakePermission } from '../../src/core/permission'
+import BlockEntity from '../../src/entities/block'
+import { DbtService } from '../../src/services/dbt'
+import { BlockParentType, BlockType } from '../../src/types/block'
+import { DbtMetadata } from '../../src/types/dbt'
+import { getConnection, getRepository } from 'typeorm'
+import _ from 'lodash'
+import { set, uuid } from '../testutils'
+import { BlockOperation } from '../../src/core/block/operation'
+
+const dbtService = new DbtService(new FakePermission())
+
+test.before(async () => {
+  await createDatabaseCon()
+  register(BlockType.DBT, DbtBlock)
+  register(BlockType.QUESTION, QuestionBlock)
+})
+
+function generateDbtBlock(metadata: DbtMetadata) {
+  const id = nanoid()
+  const entity = new BlockEntity()
+  Object.assign(entity, {
+    id,
+    interKey: id,
+    workspaceId: 'test',
+    storyId: 'test',
+    parentId: 'test',
+    parentTable: BlockParentType.WORKSPACE,
+    type: BlockType.DBT,
+    content: {
+      ...metadata,
+      title: [[`dbt.${metadata.name}`]],
+    },
+    children: [],
+    alive: true,
+  })
+  return entity
+}
+
+function generateQuestionOp(title: string, sql: string): [string, any] {
+  const id = nanoid()
+  return [
+    id,
+    {
+      id,
+      type: BlockType.QUESTION,
+      storyId: 'test',
+      parentId: 'test',
+      parentTable: BlockParentType.BLOCK,
+      content: {
+        title: [[title]],
+        sql: sql,
+      },
+      alive: true,
+      createdById: uuid(),
+    },
+  ]
+}
+
+const metadata_source_event: DbtMetadata = {
+  name: 'event',
+  description: 'just event table',
+  relationName: 'test.event',
+  compiledSql: '?',
+  type: 'source',
+  materialized: 'table',
+  sourceTable: 'event',
+}
+
+const metadata_source_user: DbtMetadata = {
+  name: 'user',
+  description: 'user table',
+  relationName: 'test.user',
+  compiledSql: '?',
+  type: 'source',
+  materialized: 'table',
+  sourceTable: 'user',
+}
+
+const metadata_source_order: DbtMetadata = {
+  name: 'order',
+  description: 'order table',
+  relationName: 'test.order',
+  compiledSql: '?',
+  type: 'source',
+  materialized: 'table',
+  sourceTable: 'order',
+}
+
+const metadata_model_dau: DbtMetadata = {
+  name: 'dau',
+  description: 'dau',
+  relationName: '',
+  rawSql: `select dt as dt, count(distinct distinct_id) as dau
+  from
+  	(select distinct_id, dt from {{ ref("event") }}) e
+    join
+	(select uid from {{ ref("user") }}) u
+	on u.uid = e.distinct_id`,
+  compiledSql: `select dt as dt, count(distinct distinct_id) as dau
+  from
+  	(select distinct_id, dt from test.events) e
+    join
+	(select uid from test.user) u
+	on u.uid = e.distinct_id`,
+  type: 'model',
+  materialized: 'table',
+}
+
+test.serial('list current dbt blocks', async (t) => {
```

Comment:
`serial` is not required. To avoid dbt blocks conflict in workspace, you can use a random workspaceId every time executing a test case

```suggestion
test('list current dbt blocks', async (t) => {
```